services/QuillLMS/app/controllers/diagnostic_insights_controller.rb

```diff
@@ -17,6 +19,22 @@ def diagnostics_assigned
     render json: fetch_diagnostics_assigned.to_json(root: false)
   end
 
+  def complete_history
+    # TODO: make this cache smarter or eradicate the need for it by speeding up the query
+    complete_history_by_classroom_data = Rails.cache.fetch("#{COMPLETE_HISTORY_CACHE_KEY}.#{current_user.id}", expires_in: 24.hours) do
+      complete_history_by_classroom = {}
+      current_user.classrooms_i_teach.each do |classroom|
+        complete_history_by_classroom[classroom.id] = {
+          non_ell: DiagnosticInsights::HistoryByClassroom.run(classroom.id, ell: false, start_date:, end_date:),
+          ell: DiagnosticInsights::HistoryByClassroom.run(classroom.id, ell: true, start_date:, end_date:)
+        }
+      end
+      complete_history_by_classroom
```

Comment:
Then you no longer need to return `complete_history_by_classroom ` since each_with_object returns the container object.